device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -339,17 +497,49 @@ public void registerDeviceClients(Iterable<DeviceClient> deviceClients) throws I
      * be used in non-multiplexing scenarios or used by the deprecated {@link TransportClient}.
      * <p>
      * @param deviceClient The device client to unregister from this multiplexing client.
+     * @throws InterruptedException If the thread gets interrupted while waiting for the unregistration to succeed.
+     * @throws com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationTimeoutException If the unregistration takes longer than the default timeout allows.
+     * @throws MultiplexingClientException If any other Exception is thrown, it will be nested into this exception.
      */
-    public void unregisterDeviceClient(DeviceClient deviceClient) throws InterruptedException
+    public void unregisterDeviceClient(DeviceClient deviceClient) throws InterruptedException, MultiplexingClientException
+    {
+        this.unregisterDeviceClient(deviceClient, DEFAULT_UNREGISTRATION_TIMEOUT_MILLISECONDS);
+    }
+
+    /**
+     * Unregister a device client from this multiplexing client. This method may be called before or after opening the
+     * multiplexed connection.
```

Comment:
Q - unregister will do a graceful close of the transport layer, right? So "register -> open -> send telemetry -> unregister -> register -> send telemetry" should be fine?

Reply:
On that note, what happens if I call deviceClient.close() but don't unregister the client from the mux connection. Will it just hang around in the connection blocking one position? If I call deviceClient.open() after that, will it be usable again?

Reply:
I see this note below: "It cannot be used in non-multiplexing scenarios", so will deviceclient.close() throw an exception?

Reply:
> Q - unregister will do a graceful close of the transport layer, right? So "register -> open -> send telemetry -> unregister -> register -> send telemetry" should be fine?

Unregister doesn't close the multiplexing connection, it simply removes the device session associated with that device client. So that flow you outlined will work, yes

Reply:
> On that note, what happens if I call deviceClient.close() but don't unregister the client from the mux connection. Will it just hang around in the connection blocking one position? If I call deviceClient.open() after that, will it be usable again?

I have a special UnsupportedOperationException to prevent users from calling open/close on the device client itself when the client is multiplexed. The only way to open and close a multiplexed client is by registering it/unregistering it

> I see this note below: "It cannot be used in non-multiplexing scenarios", so will deviceclient.close() throw an exception?

Yep. That way users can't mix up if their client is multiplexed or not